File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/endpoint/dto/user/BookingOverviewDto.java
package at.ac.tuwien.sepm.groupphase.backend.endpoint.dto.user;

import at.ac.tuwien.sepm.groupphase.backend.entity.BookingState;
import java.math.BigDecimal;

public record BookingOverviewDto(
    Long bookingId,
    Long invoiceNumber,
    Long cancellationInvoiceNumber,
    String file,
    String title,
    String date,
    String duration,
    int ticketCount,
    BigDecimal price,
    BookingState bookingState) {}

File: frontend/src/app/dtos/booking/booking-overview.dto.ts
import { BookingState } from './booking-state';

export class BookingOverviewDto {
    constructor(
        public bookingId: number,
        public invoiceNumber: number,
        public cancellationInvoiceNumber: number,
        public file: any,
        public title: string,
        public date: string,
        public duration: string,
        public ticketCount: number,
        public price: number,
        public bookingState: BookingState,
    ) {}
}
